src/tree/FunctionTreeItemBase.ts

```diff
@@ -77,20 +101,32 @@ export abstract class FunctionTreeItemBase extends AzExtTreeItem {
 
     public abstract getKey(): Promise<string | undefined>;
 
-    public async refreshImpl(): Promise<void> {
-        if (this.config.isHttpTrigger) {
+    public async refreshImpl(_context: IActionContext, isInitializing?: boolean): Promise<void> {
+        if (!isInitializing) { // no need to refresh "_func" if we're initializing the class since it's set in the constructor
+            this._func = await this.refreshFuncEnvelope();
+        }
+
+        if (this.isHttpTrigger) {
             await this.refreshTriggerUrl();
         }
 
         await this.refreshDisabledState();
     }
 
-    private async refreshTriggerUrl(): Promise<void> {
-        const triggerUrl: url.URL = new url.URL(this.parent.parent.hostUrl);
+    public abstract refreshFuncEnvelope(): Promise<WebSiteManagementModels.FunctionEnvelope | undefined>;
 
-        const route: string = (this.config.triggerBinding && this.config.triggerBinding.route) || this.name;
-        const hostJson: IParsedHostJson = await this.parent.parent.getHostJson();
-        triggerUrl.pathname = `${hostJson.routePrefix}/${route}`;
+    private async refreshTriggerUrl(): Promise<void> {
+        const hostUrl = new url.URL(this.parent.parent.hostUrl);
+        let triggerUrl: url.URL;
+        if (this._func?.invokeUrlTemplate) {
+            triggerUrl = new url.URL(this._func?.invokeUrlTemplate);
+            triggerUrl.protocol = hostUrl.protocol; // invokeUrlTemplate seems to use the wrong protocol sometimes. Make sure it matches the hostUrl
```

Comment:
Is there any information about this?

Reply:
Nah just from my own testing. Pretty harmless thing to do even if they fix the problem